### PR TITLE
Analyses Page Refactor

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Updated the user and admin analyses listing pages to use ant.design instead of datatables.

## Related issue
#413 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [X] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [X] Tests added (or description of how to test) for any new features.
~~* [ ] User documentation updated for UI or technical changes.~~ **User documentation will be updated in the final branch.**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -73,8 +73,12 @@ When you have completed development on your issue, you can submit it to the proj
 
 **The final decision on whether to accept a pull request comes from the core IRIDA development team.**
 
+### Contributing Internationalizations
+
+IRIDA needs help supporting non-English language users!  To help internationlize IRIDA, see the [IRIDA internationalization devleoper guide](https://irida.corefacility.ca/documentation/developer/interface/i18n).  To submit internationalization files back to IRIDA, submit a pull request as outlined above.
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply CONTRIBUTING.md]: Not part of this merge request, but updated in 348a706cd0d0090450844ee49d2fc172c1897172

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysesAjaxController.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	}
+
+	/**
+	 * Delete a specific {@link AnalysisSubmission}
+	 *
+	 * @param id for the {@link AnalysisSubmission} to delete
+	 */
+	@RequestMapping("/delete")
+	@ResponseBody
+	public void deleteAnalysisSubmission(@RequestParam Long id) {
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysesAjaxController.java]: Sure, don't think we normally do this, but cannot hurt.  Added in 93a8b0d020f73869126386c1bf8864f485e77017

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysesAjaxController.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+	}
+
+	/**
+	 * Get the milliseconds between two {@link Date}s
+	 *
+	 * @param start {@link Date}
+	 * @param end   {@link Date}
+	 * @return {@link Long} milliseconds
+	 */
+	private Long getDurationInMilliseconds(Date start, Date end) {
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysesAjaxController.java]: Refactored into a utility class in 3b2801da332fa6c9841e20255ac60cd58a50e106

File: src/main/java/ca/corefacility/bioinformatics/irida/service/AnalysisSubmissionService.java [hunk 4]
```diff
@@ -253,40 +253,40 @@ public ProjectAnalysisSubmissionJoin shareAnalysisSubmissionWithProject(Analysis
 	 *
 	 * @param search      basic search string
 	 * @param name        analysis submission name
-	 * @param state       {@link AnalysisState} of the submission to search
+	 * @param state       Set of {@link AnalysisState} of the submission to search
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/AnalysisSubmissionService.java]: Sounds good.  Updated in 75c7089e4cb0f213d7b3fff8afb4cb31b65c5e80

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/dto/AnalysesFilters.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package ca.corefacility.bioinformatics.irida.ria.web.analysis.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import ca.corefacility.bioinformatics.irida.model.enums.AnalysisState;
+
+/**
+ * Represents the filters on teh Analyses Table.
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/dto/AnalysesFilters.java]: Thanks, fixed in a70d997eced69e7870209dae053295f3ab8f8f3a

File: src/main/webapp/resources/js/components/ant.design/table-search-props.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { Button, Icon, Input } from "antd";
+import { blue6 } from "../../styles/colors";
+
+function getTextSearchProps(dataIndex) {
+  let searchInput;
+  return {
+    onFilter: (value, record) => {
+      // Somethings the values can be undefined so give it a default value.
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/components/ant.design/table-search-props.js]: If I must, updated in de9bda124fab9e430051583f4b48ba85b6c52904

File: src/main/webapp/resources/js/pages/analyses/AnalysisState.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { Badge, Icon } from "antd";
+import { SPACE_XS } from "../../styles/spacing";
+import { green6 } from "../../styles/colors";
+
+/**
+ * Display the state of an analysis, if it is running show the percentage
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/analyses/AnalysisState.jsx]: Your good at this, removed in 2256f5b1e6db5429837a84cdb9e87c74a1dc52d6

File: src/main/java/ca/corefacility/bioinformatics/irida/service/AnalysisSubmissionService.java [hunk 8]
```diff
@@ -253,40 +254,40 @@ public ProjectAnalysisSubmissionJoin shareAnalysisSubmissionWithProject(Analysis
...
+	public Page<AnalysisSubmission> listSubmissionsForProject(String search, String name, Set<AnalysisState> states,
 			Set<UUID> workflowIds, Project project, PageRequest pageRequest);
 
 	/**
 	 * Get a page of all {@link AnalysisSubmission}s in the system
 	 *
 	 * @param search      basic search string
 	 * @param name        analysis submission name
-	 * @param state       {@link AnalysisState} of the submission to search
+	 * @param state       Set of {@link AnalysisState} of the submission to search
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/service/AnalysisSubmissionService.java]: `state` should be changed to `states` here as well

Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/AnalysisSubmissionService.java]: :) Okay, updated in af95b2042393711b211308c7ae02dd921fa7fa59

**Pull request merged**